model/src/main/scala/org/broadinstitute/dsde/rawls/model/expressions/OutputExpression.scala

```diff
@@ -0,0 +1,56 @@
+package org.broadinstitute.dsde.rawls.model.expressions
+
+import org.broadinstitute.dsde.rawls.{RawlsException, StringValidationUtils}
+import org.broadinstitute.dsde.rawls.model.{Attribute, AttributeName, ErrorReportSource}
+
+import scala.util.{Failure, Success, Try}
+
+sealed trait OutputExpression
+case object BlankOutputExpression extends OutputExpression {
+  override def toString: String = ""
+}
+case class TargetedOutputExpression(target: ExpressionTarget, attributeName: AttributeName) extends OutputExpression {
+  override def toString: String = s"${target.root}${AttributeName.toDelimitedName(attributeName)}"
+}
+
+sealed trait OutputExpressionBinding
+case object UnboundOutputExpression extends OutputExpressionBinding
+case class BoundOutputExpression(target: ExpressionTarget, attributeName: AttributeName, attribute: Attribute) extends OutputExpressionBinding
```

Comment:
can you talk some about these classes? the difference between building and binding? what they represent and their flow? i'm not super following here

Reply:
+1